llmfoundry/utils/config_utils.py

```diff
@@ -178,9 +183,102 @@ def log_config(cfg: DictConfig) -> None:
             wandb.config.update(om.to_container(cfg, resolve=True))
 
     if 'mlflow' in cfg.get('loggers', {}):
-        try:
-            import mlflow
-        except ImportError as e:
-            raise e
+        if not mlflow:
+            raise ImportError('MLflow is required but not installed.')
         if mlflow.active_run():
             mlflow.log_params(params=om.to_container(cfg, resolve=True))
+            _log_dataset_uri(cfg)
+
+
+def _parse_source_dataset(cfg: DictConfig) -> Set[Tuple[str, str, str]]:
+    """Parse a run config for dataset information.
+
+    Given a config dictionary, parse through it to determine what the datasource
+    should be catagorized as. Possible data sources are Delta Tables, UC Volumes,
+    HuggingFace paths, remote storage, or local storage.
+
+    Args:
+        cfg (DictConfig): A config dictionary of a run
+
+    Returns:
+        Set[Tuple[str]]: A set of tuples formatted as (data type, path, split)
+    """
+    data_paths = set()
+
+    for data_split in ['train', 'eval']:
+        data_set = cfg.get(f'{data_split}_loader', {}).get('dataset', {})
```

Comment:
eval loader could be a list, probably want to handle that case (https://github.com/mosaicml/llm-foundry/pull/603)